pandas/core/algorithms.py

```diff
@@ -988,7 +988,9 @@ def _get_data_algo(values, func_map):
     elif is_unsigned_integer_dtype(values):
         f = func_map['uint64']
         values = _ensure_uint64(values)
-
+    elif is_categorical(values) and values._ordered:
```

Comment:
the problem here is that the ranking function will change based on whether the categorical is ordered. ordered categorical will be ranked based on the codes('float64') unordered categorical will be ranked as objects/strings.  This ties in with your comment on pandas/core/categoricals as well. What you showed is
````
if self.ordered:
    values = self.codes
    mask = values == -1
    if mask.any():
        values = values.astype('float64')
        values[mask] = np.nan
else:
    values = np.array(values) # values not defined inside np.array
````
values is not defined in the else clause (inside np.array) so i assume it one of self.codes or self.
In the case that you pass self.codes the ranking will not be consistent with the current functionality. In the case you pass self, a single ranking function will not be able to rank ordered/unordered categoricals. Please let me know if i'm missing something obvious here.